lib/Sources/BluetoothAction/ReadDescriptor.swift

```diff
@@ -55,4 +57,10 @@ struct ReadDescriptor: BluetoothAction {
         let event = try await client.descriptorRead(peripheral, characteristic: characteristic, descriptor: descriptor)
         return ReadDescriptorResponse(data: event.data)
     }
+
+    private func checkSecurityList(securityList: SecurityList) throws {
```

Comment:
Probably isn't worth it since they mostly have somewhat different implementations, but you could make a `Blockable` protocol and default it with the most common implementations for the two functions.

Reply:
Going to stick with the simpler if somewhat duplicative approach as there are only two cases and probably won't ever be any more.